### PR TITLE
fix: broken auth on local envinronment

### PR DESCRIPTION
**Pull request by soofstad**
## Why is this pull request needed?

- Should be able to run the web app with and without authentication in local environment 

## What does this pull request change?

- fix missing redirect url for login
- use AuthProvider state to check if loginInProgres, and conditionally don't render any components
- Update Header with more sensable user info when none available

### CODE REVIEW

File: web/src/App.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { AuthProvider } from 'react-oauth2-code-pkce'
 import { authConfig } from './authConfig'
 import { MainPage } from './pages/Main'
 
-const AUTH_DISABLED = process.env.REACT_APP_AUTH !== '1'
+export const AUTH_DISABLED = process.env.REACT_APP_AUTH !== '1'
```
Comment (olavis) [inline web/src/App.tsx:5]: I'm not sure if App.tsx is the best owner of this constant, now that it is used globally. Maybe we could store constants in a separate file, so we'll know where to find them?

File: web/.eslintignore [hunk 2]
```diff
@@ -3,4 +3,3 @@ web/src/api/generated
 
 build
 coverage
-web/src/components/TodoApp.tsx
```
Comment (olavis) [inline web/.eslintignore:6]: Good that you removed this. Next time, maybe we should do it in a separate commit to keep order in the git history. I don't see why this change belongs under the title `fix: broken auth under local environment`

File: web/src/components/common/Header.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { TopBar, Icon, Popover, Button } from '@equinor/eds-core-react'
 import { account_circle, grid_on, info_circle } from '@equinor/eds-icons'
 import { useContext, useRef, useState } from 'react'
 import { AuthContext } from 'react-oauth2-code-pkce'
+import { AUTH_DISABLED } from '../../App'
```
Comment (olavis) [inline web/src/components/common/Header.tsx:6]: Again, not sure if this constant should be exported from App.tsx. I find it hard to agree that Header is a common component now that it is dependent on another (non-common) component.

**Review by olavis — CHANGES_REQUESTED** [review]

File: web/yarn.lock [hunk 4]
```diff
@@ -10516,12 +10516,12 @@ __metadata:
...
-  version: 1.4.0
-  resolution: "react-oauth2-code-pkce@npm:1.4.0"
+"react-oauth2-code-pkce@npm:^1.7.1":
+  version: 1.7.1
+  resolution: "react-oauth2-code-pkce@npm:1.7.1"
   peerDependencies:
-    react: ^17.0.2
-  checksum: 3cb2d7126926c8538f101e03b5cee853b555692b41daef751e1509a99c582400a6c352099dc2fe502f7521520c0bad47eba31dd6bf57b114061380969f55144b
+    react: ">=16.8.0"
+  checksum: f841c1ce49b37e90b35919dc8ff207ccb10bab0d08c6ac9040ff6874b4c53207eaf41201db78c191247deaaa4ee9edc5470df200c693cd8aa433355e616cc87b
```
[the remark this answers is not in the document]
Comment (soofstad) [reply web/yarn.lock:10524]: No, it will fix that as well. But that was a side effect. The reason for the upgrade was for the "loginInProgress" feature

File: web/src/pages/Main.tsx [hunk 5]
```diff
@@ -21,6 +23,8 @@ export const MainPage = (): JSX.Element => {
             related to the app.
           </p>
         </div>
+      ) : (
+        <div>This is my app</div>
```
[the remark this answers is not in the document]
Comment (soofstad) [reply web/src/pages/Main.tsx:27]: This has not been changed in this PR. Ether way it's a placeholder that will be replaced soon.

File: web/src/components/common/Header.tsx (hunk 3 above)
Comment (soofstad) [reply web/src/components/common/Header.tsx:6]: Agreed. Header is not "common". It's only used one place.
But out of scope for this PR/issue

File: web/src/pages/Main.tsx [hunk 6]
```diff
@@ -1,15 +1,17 @@
 import { Header } from '../components/common/Header'
 import { useContext } from 'react'
 import { AuthContext } from 'react-oauth2-code-pkce'
+import { AUTH_DISABLED } from '../App'
 
 export const MainPage = (): JSX.Element => {
-  const { token } = useContext(AuthContext)
+  const { token, loginInProgress } = useContext(AuthContext)
+
+  if (loginInProgress) return <></>
```
[the remark this answers is not in the document]
Comment (soofstad) [reply web/src/pages/Main.tsx:9]: No-can-do. We need at least one level of components between the AuthProvider the the "useContext".
Could of course add another component between App and Main. But as long as we only have one page. This seems like the best solution.

File: web/src/pages/Main.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (olavis) [reply web/src/pages/Main.tsx:27]: Well, the line _has_ changed in this PR. Maybe I should remind you that it was addressed in the PR where it was introduced as well: https://github.com/equinor/mercury/pull/16 The conversation was resolved, but it doesn't seem like anything was done? And no reply? Am I supposed to just shut my mouth if I recognise issues I earlier have pointed out that has been ignored?

File: web/src/components/common/Header.tsx (hunk 3 above)
Comment (olavis) [reply web/src/components/common/Header.tsx:6]: The change was introduced in this PR, so if it's out of scope, well then changing the `AUTH_DISABLED` constant to a global is out of scope as well. The Header wasn't dependent on App.tsx until now.

File: web/src/App.tsx (hunk 1 above)
Comment (soofstad) [reply web/src/App.tsx]: Fixed

File: web/src/pages/Main.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (olavis) [reply web/src/pages/Main.tsx:9]: I'll see if I can provide an example

File: web/src/pages/Main.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (soofstad) [reply web/src/pages/Main.tsx]: No. Oversight in #16.

File: web/src/constants.ts [hunk 7]
```diff
@@ -0,0 +1 @@
+export const AUTH_DISABLED = process.env.REACT_APP_AUTH !== '1'
```
Comment (olavis) [inline web/src/constants.ts:1]: 👍

**Review by mariuswinger — APPROVED** [review]
